File: Android/SayHi/app/src/main/java/de/oklemenz/sayhi/model/NewTag.java
package de.oklemenz.sayhi.model;

import java.io.Serializable;

import de.oklemenz.sayhi.service.Utilities;

/**
 * Created by Oliver Klemenz on 22.02.17.
 */

public class NewTag implements Serializable {

    public boolean assignPos = true;

    public String key = "";
    public String langCode = "";
    private String name = "";

    public void setName(String name) {
        this.name = Utilities.capitalize(name);
    }

    public String getName() {
        return name;
    }

    public Category category;
    public boolean categoryNew;
    public Tag primaryLangTag;
    public Category primaryLangCategory;

    public Tag getTag() {
        Tag tag = new Tag(
                key,
                langCode,
                name,
                category.key,
                primaryLangTag != null ? primaryLangTag.key : "");
        tag.category = category;
        return tag;
    }
}

File: iOS/SayHi/SayHi/NewTag.swift
//
//  NewTag.swift
//  SayHi
//
//  Created by Klemenz, Oliver on 19.12.16.
//  Copyright © 2016 Klemenz, Oliver. All rights reserved.
//

import Foundation

class NewTag {

    var assignPos: Bool = true
    
    var key: String = ""
    var langCode: String = ""
    var _name: String = ""
    var name: String {
        set {
            _name = newValue.capitalize
        }
        get { return _name }
    }
    var category: Category?
    var categoryNew: Bool = false
    var primaryLangTag: Tag?
    var primaryLangCategory: Category?
    
    var tag : Tag {
        let tag = Tag(key: key,
                      langCode: langCode,
                      name: name,
                      categoryKey: category!.key,
                      primaryLangKey: primaryLangTag?.key ?? "")
        if let category = self.category {
            tag._category = category
        }
        return tag
    }
}
